src/python/pants/option/global_options.py

```diff
@@ -993,6 +993,29 @@ class BootstrapOptions:
             """
         ),
     )
+    native_options_validation = EnumOption(
+        default=NativeOptionsValidation.WARNING,
+        help=softwrap(
+            """
+            Pants is switching its option parsing system from a legacy parser written in Python
+            to a new one written in Rust.
+
+            The results of parsing a given option by each system should be identical. However
+            during a transition period we will run both parsers and compare their results.
+            This option controls how to report discrepancies that arise.
+
+            ERROR: Discrepancies will cause Pants to exit.
+            WARNING: Discrepancies will be logged but Pants will continue.
+            IGNORE: A last resort to turn off this check entirely.
```

Comment:
Nice approach. 

Very minor, but I note that:

-  most of this sort of enums use lower case values, e.g. `NoApplicableTargetsBehavior` or `GlobMatchErrorBehavior`.
- I wonder whether this will render all on one line in Markdown on the docs site, and so maybe could be a list?

```suggestion
            - `error`: Discrepancies will cause Pants to exit.
            - `warning`: Discrepancies will be logged but Pants will continue.
            - `ignore`: A last resort to turn off this check entirely.
```